### ISSUE TITLE
Option to make messaging studies show once or re-prompt 

### ISSUE THREAD

**Issue by Osmose** [mozilla/normandy#336]
#347 changes the `show-heartbeat` action to only prompt the user once, and never again, per-recipe. To accommodate the long-term Release Heartbeat survey, in which we do want to occasionally re-prompt users (after a long time we'll exhaust the user population, so we want to eventually make people eligible to see the prompt again), we should add a new argument to `show-heartbeat` recipes for configuring when a recipe should re-prompt users.

The new argument should be a choice between two options, using radio buttons:

- Do not show this prompt to specific users more than once
- Allow re-prompting users who have already seen this prompt after `X` days since they last saw it

`X` should be a number input that defaults to 30, is disabled if the option isn't selected, and is required if the option is selected.

When run, `show-heartbeat` should store the datetime that it was seen. If the first option is selected, the action should not prompt the user if any datetime is logged If the second option is selected, the action should compare the request time returned by the server with the stored datetime, and if the difference is larger than the specified delay, then it should re-prompt and store the request time as the last-seen date.

**Comment by MattGrimes** [comment]
We would normally only want to reprompt a user when the message is something important enough that we want them to make an explicit choice. We should stop prompting when the user clicks on a button, rates with stars, or answers a question.

**Comment by Osmose** [comment]
@MattGrimes I think we'll also want the option for re-prompting after X days regardless of interaction for the ongoing Heartbeat survey. I'm expecting it to be set to something like 90-180 days. Does that make sense to keep as a third option alongside the user-interaction-specific option?

**Comment by MattGrimes** [comment]
Absolutely if it's something you are willing to build. Hack would be to create a new recipe ID and just use the same bucket.

**Comment by Osmose** [comment]
So then the new spec is 3 options:

- Do not show this prompt to specific users more than once
- Show this prompt until the user interacts with it in any way, and then never again.
- Allow re-prompting users who have already seen this prompt after `X` days since they last saw it

**Comment by mythmon** [comment]
Does it make sense to have separate actions for this? That way we could have unique UI for the different cases, and it may simplify the code in the actions as well.

**Comment by andymikulski** [comment]
After thinking about it for a few days - there's definitely arguments for either approach.

IMO, I think the UX would be better if the `show-heartbeat` action just had these as configurations versus having a bunch of additional actions like `show-heartbeat (once)`, `show-heartbeat (nag)`, etc pollute the actions list. Though, to be fair, multiple actions would be easily filterable on the recipe list page.

In terms of LOE for either approach - they're about the same. Adding the options to the `show-heartbeat` arguments is pretty simple to do. Creating multiple actions of `show-heartbeat` is a little bit more work (abstracting the ShowHeartbeat action as much as possible to prevent repeating code) but doesn't seem daunting or anything.

@mythmon Any thoughts?

**Comment by mythmon** [comment]
My main concerns were UX and the effort to implement. Since it would be about the same difficulty to implement, and not a better UX, lets go with the original plan of making a more configurable show-heartbeat.

Thanks for thinking through the options!

**Comment by andymikulski** [comment]
>We should stop prompting when the user clicks on a button, rates with stars, or answers a question.
\- @MattGrimes 

We're able to track the following events from the heartbeat prompt:
```
Engaged,
LearnMore,
NotificationOffered,
NotificationClosed,
TelemetrySent,
Voted
```

I'm assuming we should track `Voted` and `Engaged`, but should we also count `LearnMore` as satisfying the 'user interaction' requirement?

**Comment by mythmon** [comment]
Closed by #429.
